src/main/kotlin/com/openlattice/projector/ProjectEntitySetEntryProcessor.kt

```diff
@@ -0,0 +1,114 @@
+package com.openlattice.projector
+
+import com.hazelcast.core.Offloadable
+import com.kryptnostic.rhizome.hazelcast.processors.AbstractRhizomeEntryProcessor
+import com.openlattice.ApiUtil
+import com.openlattice.edm.EntitySet
+import com.openlattice.edm.set.EntitySetFlag
+import com.openlattice.postgres.PostgresColumn
+import com.openlattice.transporter.tableName
+import com.openlattice.transporter.types.TransporterDatastore
+import com.openlattice.transporter.types.TransporterDependent
+import org.apache.olingo.commons.api.edm.FullQualifiedName
+import org.slf4j.LoggerFactory
+import java.util.*
+
+/**
+ *
+ * @author Drew Bailey &lt;drew@openlattice.com&gt;
+ */
+data class ProjectEntitySetEntryProcessor(
+        val columns: Map<UUID, FullQualifiedName>,
+        val organizationId: UUID,
+        val usersToColumnPermissions: Map<String, List<String>>
+): AbstractRhizomeEntryProcessor<UUID, EntitySet, Void?>(),
+        Offloadable,
+        TransporterDependent<ProjectEntitySetEntryProcessor>
+{
+    @Transient
+    private lateinit var data: TransporterDatastore
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(ProjectEntitySetEntryProcessor::class.java)
+    }
+
+    override fun process(entry: MutableMap.MutableEntry<UUID, EntitySet>): Void? {
+        check(::data.isInitialized) { TransporterDependent.NOT_INITIALIZED }
+        val es = entry.value
+        if ( es.flags.contains( EntitySetFlag.MATERIALIZED )){
+            return null
+        }
+        es.flags.add(EntitySetFlag.MATERIALIZED)
+        entry.setValue(es)
```

Comment:
should probably move this piece to the end of the `try` so the entity set doesn't get marked as materialized if the create view piece doesn't complete

Reply:
ah I see, it's updated in the catch. Will the `catch` be hit if prod goes down partway through the `try`?

Reply:
🧐

Reply:
I moved it to `finally {}`